readthedocs/api/v2/views/model_views.py

```diff
@@ -237,12 +252,22 @@ def get_serializer_class(self):
 
     @decorators.action(
         detail=False,
-        permission_classes=[permissions.IsAdminUser],
+        permission_classes=[HasBuildAPIKey | permissions.IsAdminUser],
         methods=['get'],
     )
     def concurrent(self, request, **kwargs):
         project_slug = request.GET.get('project__slug')
-        project = get_object_or_404(Project, slug=project_slug)
+        build_api_key = request.build_api_key
+        if build_api_key:
+            if project_slug != build_api_key.project.slug:
+                log.info(
+                    "Project slug doesn't match the one attached to the API key.",
+                    project_slug=project_slug,
```

Comment:
It would be good to log the `build_api_key.id` as well so we can relate match both.

Reply:
This is already done https://github.com/readthedocs/readthedocs.org/blob/054897e81e225829529e6b5a611c6d7cf5ba82ad/readthedocs/api/v2/views/model_views.py#L258-L262